tests/compat.sh

```diff
@@ -110,6 +111,28 @@ print_usage() {
     printf "            \tAlso available: GnuTLS (needs v3.2.15 or higher)\n"
     printf "  -M|--memcheck\tCheck memory leaks and errors.\n"
     printf "  -v|--verbose\tSet verbose output.\n"
+    printf "     --list-test-case\tList all potential test cases (No Execution)\n"
+}
+
+list_test_case() {
+    reset_ciphersuites
+    for TYPE in $TYPES; do
+        add_common_ciphersuites
+        add_openssl_ciphersuites
+        add_gnutls_ciphersuites
+        add_mbedtls_ciphersuites
+    done
+
+    for VERIFY in $VERIFIES; do
+        VERIF=$(echo $VERIFY | tr '[:upper:]' '[:lower:]')
+        for MODE in $MODES; do
+            print_test_title m o "$O_CIPHERS"
+            print_test_title o m "$O_CIPHERS"
+            print_test_title m g "$G_CIPHERS"
+            print_test_title g m "$G_CIPHERS"
+            print_test_title m m "$M_CIPHERS"
```

Comment:
The generated test case descriptions are used as keys for analyzing outcome files. They should be identical to the `$TITLE` which would be printed when run the tests.
```suggestion
            print_test_title m O "$O_CIPHERS"
            print_test_title O m "$O_CIPHERS"
            print_test_title m G "$G_CIPHERS"
            print_test_title G m "$G_CIPHERS"
            print_test_title m m "$M_CIPHERS"
```